Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -143,10 +141,31 @@ public class Service : AbstractClientService, ISubmitterService
                                     return;
                                   }
 
+                                  var distinctRetry = blockRequestList.Select(x => x.TaskOptions.MaxRetries)
+                                                                      .Distinct()
+                                                                      .Count();
+                                  var distinctPriority = blockRequestList.Select(x => x.TaskOptions.Priority)
+                                                                         .Distinct()
+                                                                         .Count();
+
+                                  if (distinctRetry > 1)
+                                  {
+                                    throw new
+                                      InvalidOperationException("The taskOptions maxRetries in AsyncBuffer has at least one different value. Please bundle tasks with same maxRetries");
+                                  }
+
+                                  if (distinctPriority > 1)
+                                  {
+                                    throw new
+                                      InvalidOperationException("The taskOptions priority in AsyncBuffer has at least one different value. Please bundle tasks with same Priority");
+                                  }
+
```

Comment:
What does this has to do with TaskOptions?

Reply:
Can't do a groupby ?